File: solutions/2026-08-06/java/2026-08-06-smallest-divisible-digit-product-i.java
```java
// Problem: Smallest Divisible Digit Product I
// Link: https://leetcode.com/problems/smallest-divisible-digit-product-i/
// Problem Summary: Find the smallest number >= n whose digit product is divisible by t.

// Approach:
// We need to find the smallest number starting from 'n' that satisfies the condition.
// The most straightforward approach is to iterate through numbers starting from 'n'
// and for each number, calculate the product of its digits. If this product is
// divisible by 't', we return that number.

// The digit product calculation involves extracting each digit of the number
// and multiplying them. If any digit is 0, the product will be 0.
// A product of 0 is divisible by any non-zero 't'.

// Time Complexity: O(X * log10(X)) where X is the smallest number >= n that satisfies the condition.
// In the worst case, n is small (e.g., 1) and t is large, we might have to iterate many numbers.
// However, since n is up to 100 and t is up to 10, the search space for X is relatively small.
// The log10(X) factor comes from calculating the digit product of each number.
// For n <= 100 and t <= 10, X won't be excessively large. The maximum value of X will be
// bounded because we are guaranteed to find a solution (e.g., a number with a digit 0
// will have a product of 0, which is divisible by any t). The problem statement implies
// that a solution always exists within reasonable bounds for these constraints.

// Space Complexity: O(1)
// We are only using a few variables to store the current number, digit product, and digits.
// The space used does not grow with the input size.

class Solution {
    /**
     * Calculates the product of the digits of a given number.
     *
     * @param num The number to calculate the digit product for.
     * @return The product of the digits of the number. Returns 0 if any digit is 0.
     */
    private int calculateDigitProduct(int num) {
        // If the number is 0, its digit product is 0.
        if (num == 0) {
            return 0;
        }

        int product = 1;
        int temp = num; // Use a temporary variable to avoid modifying the original number

        // Iterate through each digit of the number
        while (temp > 0) {
            int digit = temp % 10; // Get the last digit

            // If any digit is 0, the product will be 0, which is divisible by any t.
            // We can return 0 immediately.
            if (digit == 0) {
                return 0;
            }

            product *= digit; // Multiply the digit into the product
            temp /= 10;       // Remove the last digit
        }
        return product;
    }

    /**
     * Finds the smallest number greater than or equal to n such that the product of its digits is divisible by t.
     *
     * @param n The starting number.
     * @param t The divisor.
     * @return The smallest number satisfying the condition.
     */
    public int smallestDivisibleDigitProduct(int n, int t) {
        // Start checking from 'n' itself.
        int currentNumber = n;

        // Loop indefinitely until a number satisfying the condition is found.
        // Since n >= 1 and t >= 1, and a 0 digit will always make the product 0
        // (which is divisible by any t), a solution is guaranteed to be found.
        while (true) {
            // Calculate the product of digits for the current number.
            int digitProduct = calculateDigitProduct(currentNumber);

            // Check if the digit product is divisible by 't'.
            // Note: If digitProduct is 0, and t is non-zero, 0 % t is 0, so it's divisible.
            if (digitProduct % t == 0) {
                // If divisible, we've found the smallest such number.
                return currentNumber;
            }

            // If not divisible, increment the number and try again.
            currentNumber++;
        }
    }

    // Main method for testing purposes (optional, but good for local execution)
    public static void main(String[] args) {
        Solution sol = new Solution();

        // Example 1:
        int n1 = 10, t1 = 2;
        System.out.println("Input: n = " + n1 + ", t = " + t1);
        System.out.println("Output: " + sol.smallestDivisibleDigitProduct(n1, t1)); // Expected: 10

        // Example 2:
        int n2 = 15, t2 = 3;
        System.out.println("Input: n = " + n2 + ", t = " + t2);
        System.out.println("Output: " + sol.smallestDivisibleDigitProduct(n2, t2)); // Expected: 16

        // Additional test case
        int n3 = 1, t3 = 5;
        System.out.println("Input: n = " + n3 + ", t = " + t3);
        System.out.println("Output: " + sol.smallestDivisibleDigitProduct(n3, t3)); // Expected: 5 (digit product 5)

        // Additional test case with 0
        int n4 = 9, t4 = 10;
        System.out.println("Input: n = " + n4 + ", t = " + t4);
        System.out.println("Output: " + sol.smallestDivisibleDigitProduct(n4, t4)); // Expected: 10 (digit product 0)

        // Additional test case with higher n and t
        int n5 = 50, t5 = 7;
        System.out.println("Input: n = " + n5 + ", t = " + t5);
        System.out.println("Output: " + sol.smallestDivisibleDigitProduct(n5, t5)); // Expected: 56 (5*6=30, not divisible. 57 -> 35, divisible)
    }
}
```

File: solutions/2026-08-06/cpp/2026-08-06-smallest-divisible-digit-product-i.cpp
```cpp
// Problem: Find the smallest number >= n whose digit product is divisible by t.
// Link: https://leetcode.com/problems/smallest-divisible-digit-product-i/
// Approach:
// We will iterate through numbers starting from n. For each number, we calculate the product of its digits.
// If the digit product is divisible by t, we return that number.
// The digit product can be 0 if the number contains a 0 digit. 0 is divisible by any non-zero t.
// To calculate the digit product:
// Initialize product to 1.
// Iterate through the digits of the number by repeatedly taking the number modulo 10 to get the last digit,
// and then dividing the number by 10 to remove the last digit.
// If any digit is 0, the product becomes 0.
// Time Complexity: O(N * log10(N)), where N is the smallest number satisfying the condition.
// In the worst case, N can be slightly larger than the initial n. The log10(N) factor comes from calculating the digit product for each number.
// Given the constraints (n <= 100), N will not be excessively large.
// Space Complexity: O(1), as we are only using a few variables to store intermediate results.

#include <iostream>

// Function to calculate the product of digits of a number
long long digitProduct(int num) {
    if (num == 0) {
        return 0; // Product of digits for 0 is 0
    }
    long long product = 1;
    int temp = num;
    while (temp > 0) {
        int digit = temp % 10;
        if (digit == 0) {
            return 0; // If any digit is 0, the product is 0
        }
        product *= digit;
        temp /= 10;
    }
    return product;
}

// Main function to find the smallest number
int smallestDivisibleDigitProduct(int n, int t) {
    int currentNum = n;
    while (true) {
        long long prod = digitProduct(currentNum);
        // Check if the product is divisible by t.
        // If t is 1, any digit product is divisible by 1.
        // If prod is 0, it's divisible by any non-zero t.
        if (t == 1 || (prod != 0 && prod % t == 0) || (prod == 0 && t != 0)) {
            return currentNum;
        }
        currentNum++; // Move to the next number
    }
}

/*
// Example Usage (for testing purposes, not part of the LeetCode submission)
int main() {
    // Example 1
    int n1 = 10, t1 = 2;
    std::cout << "Input: n = " << n1 << ", t = " << t1 << std::endl;
    std::cout << "Output: " << smallestDivisibleDigitProduct(n1, t1) << std::endl; // Expected: 10

    // Example 2
    int n2 = 15, t2 = 3;
    std::cout << "Input: n = " << n2 << ", t = " << t2 << std::endl;
    std::cout << "Output: " << smallestDivisibleDigitProduct(n2, t2) << std::endl; // Expected: 16

    // Additional Test Cases
    int n3 = 1, t3 = 1;
    std::cout << "Input: n = " << n3 << ", t = " << t3 << std::endl;
    std::cout << "Output: " << smallestDivisibleDigitProduct(n3, t3) << std::endl; // Expected: 1

    int n4 = 5, t4 = 7;
    std::cout << "Input: n = " << n4 << ", t = " << t4 << std::endl;
    std::cout << "Output: " << smallestDivisibleDigitProduct(n4, t4) << std::endl; // Expected: 7

    int n5 = 99, t5 = 10;
    std::cout << "Input: n = " << n5 << ", t = " << t5 << std::endl;
    std::cout << "Output: " << smallestDivisibleDigitProduct(n5, t5) << std::endl; // Expected: 100 (prod=0)

    return 0;
}
*/
```